### PR TITLE
[BEAM-8876] Run hadoop tests with different versions and enable HadoopFormatIT test on Postcommit

### PR DESCRIPTION
**Pull request by piotr-szuberski**
- Enable HadoopFormatIOIT test with postgres container
- Add multiple versions of Hadoop to tests

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run Spark ValidatesRunner

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run Spark ValidatesRunner

**Comment by piotr-szuberski** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by piotr-szuberski** [comment]
@timrobertson100 @iemejia Could I ask you to take a look at this PR? I screwed up a bit the previous PR (update Hadoop to 2.10.1 and test with 2.8.5) because I didn't change the classpath variable which is essential. The Gradle magic :)

**Comment by piotr-szuberski** [comment]
Run Java PreCommit

**Comment by timrobertson100** [comment]
Thanks @piotr-szuberski 

I've asked @iemejia in a chat for advice on how we should proceed (either fix this here which I think makes sense, or revert the last PR and fix in one PR)

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run Java PreCommit

**Comment by piotr-szuberski** [comment]
Run Python PreCommit

**Comment by piotr-szuberski** [comment]
Run Java PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by piotr-szuberski** [comment]
Run Spark ValidatesRunner

**Comment by piotr-szuberski** [comment]
Run Java PreCommit

**Comment by piotr-szuberski** [comment]
Run Java PreCommit

**Review by iemejia — APPROVED** [review]
LGTM, Thanks for the work on this, it is really great to see this happening, and thanks for fixing the classpath issue, it was relatively easy to miss so :+1:.

As a follow up it would be extremely nice to have also:
1. A phrase based task to invoke the Hadoop Versions tests from a PR e.g. "Run Hadoop Versions Compatibility Tests"
2. Make the hadoopTests run by default on precommit if the hadoop related modules are touched. (triggerPathPatterns)

File: sdks/java/io/hadoop-common/build.gradle [hunk 1]
```diff
@@ -16,17 +16,21 @@
...
 description = "Apache Beam :: SDKs :: Java :: IO :: Hadoop Common"
 ext.summary = "Library to add shared Hadoop classes among Beam IOs."
 
-configurations {
-  hadoopVersion285
-}
+def hadoopVersions = [
+    "285": "2.8.5",
+    "292": "2.9.2",
+    "2101": "2.10.1",
```
[the remark this answers is not in the document]
Comment (piotr-szuberski) [reply sdks/java/io/hadoop-common/build.gradle:30]: I have the same issue - I tried to add some global variable to BeamModulePlugin but I don't get the Gradle's keikaku how to do such thing.

About 3.2.1 - there are Jira tasks for it so I wanted to do it in another PR (as this one already is about 2 things at once - IT test and version testing). I'm not sure whether every module passes the tests. I'll do that right after this is merged.

**Comment by piotr-szuberski** [comment]
> As a follow up it would be extremely nice to have also:
> 
>     1. A phrase based task to invoke the Hadoop Versions tests from a PR e.g. "Run Hadoop Versions Compatibility Tests"
> 
>     2. Make the hadoopTests run by default on precommit if the hadoop related modules are touched. (triggerPathPatterns)

Will do!

**Pull request merged**